flink-connectors/flink-connector-kafka-base/src/test/java/org/apache/flink/streaming/connectors/kafka/KafkaJsonTableSourceFactoryTestBase.java

```diff
@@ -89,9 +94,10 @@ private void testTableSource(FormatDescriptor format) {
 		// construct table source using a builder
 
 		final Map<String, String> tableJsonMapping = new HashMap<>();
+		tableJsonMapping.put("name", "name");
```

Comment:
This "name" to "name" mapping should not exist since we've already explicitly defined the "fruit-name" to "name" mapping.

Reply:
The question is what should a rowtime attribute field (or a custom extractor) reference? The input or the current schema? I think it should reference the input thus all fields (even the renamed ones) need to be present in the mapping.

Reply:
Well, according to the current implementation, you are right. But I still feel uncomfortable about that since we actually mix the physical schema (format schema) and the logical schema (table schema) into the same map. Do you think it's necessary to make some changes here?

Reply:
@fhueske what do you think about this whole mapping business?

Reply:
```
physical schema ==mapping=> "intermediate schema" ==timestamp extraction and projection=> logical schema
```
Maybe we should consider eliminating the "intermedia schema" in the future?